includes/model/class-follower.php

```diff
@@ -0,0 +1,258 @@
+<?php
+namespace Activitypub\Model;
+
+use Activitypub\Collection\Followers;
+
+/**
+ * ActivityPub Follower Class
+ *
+ * This Object represents a single Follower.
+ * There is no direct reference to a WordPress User here.
+ *
+ * @author Matthias Pfefferle
+ *
+ * @see https://www.w3.org/TR/activitypub/#follow-activity-inbox
+ */
+class Follower {
+	/**
+	 * The Object ID
+	 *
+	 * @var int
+	 */
+	private $id;
+
+	/**
+	 * The Actor-URL of the Follower
+	 *
+	 * @var string
+	 */
+	private $actor;
+
+	/**
+	 * The Object slug
+	 *
+	 * This is a requirement of the Term-Meta but will not
+	 * be actively used in the ActivityPub context.
+	 *
+	 * @var string
+	 */
+	private $slug;
+
+	/**
+	 * The Object Name
+	 *
+	 * This is the same as the Actor-URL
+	 *
+	 * @var string
+	 */
+	private $name;
+
+	/**
+	 * The Username
+	 *
+	 * @var string
+	 */
+	private $username;
+
+	/**
+	 * The Avatar URL
+	 *
+	 * @var string
+	 */
+	private $avatar;
+
+	/**
+	 * The URL to the Followers Inbox
+	 *
+	 * @var string
+	 */
+	private $inbox;
+
+	/**
+	 * The URL to the Servers Shared-Inbox
+	 *
+	 * If the Server does not support Shared-Inboxes,
+	 * the Inbox will be stored.
+	 *
+	 * @var string
+	 */
+	private $shared_inbox;
+
+	/**
+	 * The date, the Follower was updated
+	 *
+	 * @var string untixtimestamp
+	 */
+	private $updated_at;
+
+	/**
+	 * The complete Remote-Profile of the Follower
+	 *
+	 * @var array
+	 */
+	private $meta;
+
+	/**
+	 * Maps the meta fields to the local db fields
+	 *
+	 * @var array
+	 */
+	private $map_meta = array(
+		'name'              => 'name',
+		'preferredUsername' => 'username',
+		'inbox'             => 'inbox',
+	);
+
+	/**
+	 * Constructor
+	 *
+	 * @param WP_Post $post
+	 */
+	public function __construct( $actor ) {
+		$this->actor = $actor;
+
+		$term = get_term_by( 'name', $actor, Followers::TAXONOMY );
+
+		if ( $term ) {
+			$this->id = $term->term_id;
+			$this->slug = $term->slug;
+			$this->meta = json_decode( $term->meta );
+		}
+	}
+
+	/**
+	 * Magic function to implement getter and setter
+	 *
+	 * @param string $method
+	 * @param string $params
+	 *
+	 * @return void
+	 */
+	public function __call( $method, $params ) {
+		$var = \strtolower( \substr( $method, 4 ) );
+
+		if ( \strncasecmp( $method, 'get', 3 ) === 0 ) {
+			if ( empty( $this->$var ) ) {
+				return $this->get( $var );
+			}
+			return $this->$var;
+		}
+
+		if ( \strncasecmp( $method, 'set', 3 ) === 0 ) {
+			$this->$var = $params[0];
+		}
+	}
+
+	public function from_meta( $meta ) {
+		$this->meta = $meta;
+
+		foreach ( $this->map_meta as $remote => $internal ) {
+			if ( ! empty( $meta[ $remote ] ) ) {
```

Comment:
This (and all `$meta['...']` notations below) causes a fatal when `$meta` is a `WP_Error`.